pkgs/build-support/rust/default.nix

```diff
@@ -17,7 +17,7 @@
   # Please set to true on any Rust package updates. Once all packages set this
   # to true, we will delete and make it the default. For details, see the Rust
   # section on the manual and ./README.md.
-, legacyCargoFetcher ? true
+, legacyCargoFetcher ? false
```

Comment:
This is the actual change on the PR.  Once merged I'll send another one updating the manual and deleting this doc comment.

Since the PR is so big and likely to generate merge conflicts as others submit PRs bumping Rust applications, for now I've just stuck to a diff that can be programmatically regenerated after a hard reset to origin/master, though I could make the sed smarter.